packages/core/src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -44,12 +52,70 @@ function setup(options: ReactNativeErrorHandlersOptions): void {
  * Setup unhandled promise rejection tracking
  */
 function setupUnhandledRejectionsTracking(patchGlobalPromise: boolean): void {
-  if (patchGlobalPromise) {
-    polyfillPromise();
-  }
+  try {
+    if (isHermesEnabled() && RN_GLOBAL_OBJ.HermesInternal?.enablePromiseRejectionTracker) {
+      logger.log('Using Hermes native promise rejection tracking');
+
+      RN_GLOBAL_OBJ.HermesInternal.enablePromiseRejectionTracker({
+        allRejections: true,
+        onUnhandled: (id: string, error: unknown) => {
+          if (__DEV__) {
+            logger.warn(`Possible Unhandled Promise Rejection (id: ${id}):\n${error}`);
+          }
+
+          // Marking the rejection as handled to avoid breaking crash rate calculations.
+          // See: https://github.com/getsentry/sentry-react-native/issues/4141
+          captureException(error, {
+            data: { id },
+            originalException: error,
+            syntheticException: isErrorLike(error) ? undefined : createSyntheticError(),
+            mechanism: { handled: true, type: 'onunhandledrejection' },
+          });
```

Comment:
Since the handling logic is the same for Hermes Promise and JSC Promise Polyfill I think we should share the impl to avoid discrepancies.

Reply:
Good point 👍 
Updated with 4ed5c880ff10f40a566a4c16915760244da91367 to share the handling logic